modin/pandas/base.py

```diff
@@ -3127,6 +3127,42 @@ def __init__(
         ]
         self.__groups = self.__get_groups(*self.resample_args)
 
+    def __getitem__(self, key):
+        """
+        Get ``Resampler`` based on `key` columns of original dataframe.
+
+        Parameters
+        ----------
+        key : str or list
+            String or list of selections.
+
+        Returns
+        -------
+        modin.pandas.BasePandasDataset
+            New ``Resampler`` based on `key` columns subset
+            of the original dataframe.
+        """
+
+        def _get_new_resampler(key):
```

Comment:
I wonder if it's a correct way - to always generate a new resampler instead of re-using an already existing one...

Reply:
Pandas also work. Specifically, when calling this function, we do not change the current resampler, but return the elements by key, formed into a new resampler.